js/bundle_test.go

```diff
@@ -404,6 +410,141 @@ func TestNewBundleFromArchive(t *testing.T) {
 	assert.Equal(t, "hi!", v2.Export())
 }
 
+func TestOpen(t *testing.T) {
+	var testCases = [...]struct {
+		name           string
+		openPath       string
+		pwd            string
+		isError        bool
+		isArchiveError bool
+	}{
+		{
+			name:     "notOpeningUrls",
+			openPath: "github.com",
+			isError:  true,
+			pwd:      "/path/to",
+		},
+		{
+			name:     "simple",
+			openPath: "file.txt",
+			pwd:      "/path/to",
+		},
+		{
+			name:     "simple with dot",
+			openPath: "./file.txt",
+			pwd:      "/path/to",
+		},
+		{
+			name:     "simple with two dots",
+			openPath: "../to/file.txt",
+			pwd:      "/path/not",
+		},
+		{
+			name:     "fullpath",
+			openPath: "/path/to/file.txt",
+			pwd:      "/path/to",
+		},
+		{
+			name:     "fullpath2",
+			openPath: "/path/to/file.txt",
+			pwd:      "/path",
+		},
+		{
+			name:     "file is dir",
+			openPath: "/path/to/",
+			pwd:      "/path/to",
+			isError:  true,
+		},
+		{
+			name:     "file is missing",
+			openPath: "/path/to/missing.txt",
+			isError:  true,
+		},
+	}
+	fss := map[string]func() (afero.Fs, string, func()){
+		"MemMapFS": func() (afero.Fs, string, func()) {
+			fs := afero.NewMemMapFs()
+			require.NoError(t, fs.MkdirAll("/path/to", 0755))
+			require.NoError(t, afero.WriteFile(fs, "/path/to/file.txt", []byte(`hi`), 0644))
+			return fs, "", func() {}
+		},
+		"OsFS": func() (afero.Fs, string, func()) {
+			prefix, err := ioutil.TempDir("", "k6_open_test")
+			require.NoError(t, err)
+			fs := afero.NewOsFs()
+			require.NoError(t, fs.MkdirAll(filepath.Join(prefix, "/path/to"), 0755))
+			require.NoError(t, afero.WriteFile(fs, filepath.Join(prefix, "/path/to/file.txt"), []byte(`hi`), 0644))
+			return fs, prefix, func() { os.RemoveAll(prefix) }
+		},
+	}
+
+	for name, fsInit := range fss {
+		fs, prefix, cleanUp := fsInit()
+		defer cleanUp()
+		fs = afero.NewReadOnlyFs(fs)
+		t.Run(name, func(t *testing.T) {
+			fs = afero.NewReadOnlyFs(fs)
+			for _, tCase := range testCases {
+				tCase := tCase
+
+				var testFunc = func(t *testing.T) {
+					var openPath = tCase.openPath
+					// if fullpath prepend prefix
+					if openPath[0] == '/' || openPath[0] == '\\' {
+						openPath = filepath.Join(prefix, openPath)
+					}
+					if runtime.GOOS == "windows" {
+						openPath = strings.Replace(openPath, `\`, `\\`, -1)
+					}
+					src := &lib.SourceData{
+						Filename: filepath.Join(prefix, "/path/to/script.js"),
+						Data: []byte(`
+			export let file = open("` + openPath + `");
+			export default function() { return file };
+		`),
+					}
+					sourceBundle, err := NewBundle(src, fs, lib.RuntimeOptions{})
+					if tCase.isError {
+						assert.Error(t, err)
+						return
+					}
+					if !assert.NoError(t, err) {
```

Comment:
I think `require.NoError()` could be used instead of the `if !assert.NoError(t, err) { return }` here and in a lot of places below